File: core/src/main/java/heroes/journey/utils/input/Options.java
package heroes.journey.utils.input;

import static heroes.journey.registries.Registries.ActionManager;

import java.util.ArrayList;
import java.util.List;

import heroes.journey.entities.actions.Action;
import heroes.journey.entities.actions.TeamActions;
import heroes.journey.entities.actions.inputs.ActionInput;
import heroes.journey.entities.actions.options.BooleanOptionAction;
import heroes.journey.entities.actions.options.OptionAction;
import heroes.journey.entities.actions.results.ActionResult;
import heroes.journey.ui.HUD;
import heroes.journey.ui.hudstates.ActionSelectState;

public class Options {

    private static List<Action> optionsList = new ArrayList<>(2);

    public static void addOption(OptionAction option) {
        option.setDisplay("");
        optionsList.add(option);
    }

    public static boolean isTrue(String optionAction) {
        Action action = ActionManager.get(optionAction);
        if (action instanceof BooleanOptionAction booleanOptionAction) {
            return booleanOptionAction.isTrue();
        }
        return false;
    }

    public static void toggle(String optionAction) {
        Action action = ActionManager.get(optionAction);
        if (action instanceof BooleanOptionAction booleanOptionAction) {
            booleanOptionAction.onSelect(null, false);
        }
    }

    static {
        Action optionsAction = new Action("options", "Options", "", true, null) {
            @Override
            public ActionResult internalOnSelect(ActionInput input) {
                HUD.get().setState(new ActionSelectState(optionsList));
                return null;
            }
        };
        TeamActions.addTeamAction(optionsAction);
    }
}

File: core/src/main/java/heroes/journey/utils/input/Options.kt
package heroes.journey.utils.input

import heroes.journey.GameState
import heroes.journey.entities.actions.ActionContext
import heroes.journey.entities.actions.ActionUtils
import heroes.journey.entities.actions.options.BooleanOptionAction
import heroes.journey.modlib.actions.ActionEntry
import heroes.journey.modlib.actions.IAction
import heroes.journey.mods.Registries

object Options {
    val optionsList: MutableList<ActionEntry> = ArrayList(2)

    fun addOption(option: IAction) {
        optionsList.add(ActionEntry(option.id, emptyMap()))
    }

    fun isTrue(optionAction: String?): Boolean {
        val action = Registries.ActionManager[optionAction]
        if (action is BooleanOptionAction) {
            return action.isTrue
        }
        return false
    }

    fun toggle(optionAction: String?) {
        val action = Registries.ActionManager[optionAction]
        if (action is BooleanOptionAction) {
            ActionUtils.onSelect(
                action,
                ActionContext(
                    GameState.global(), false
                )
            )
        }
    }
}
